### ISSUE TITLE
[rllib] Usability improvements

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#2053]
### Describe the problem

This is a placeholder for potential rllib usability improvements:
- [x] Add accessor methods for reading and updating policy / preprocessor state at runtime. This allows more complex training workflows to be expressed in RLlib.
- [x] Support user defined policies (which should be just tensorflow models that define action and loss output tensors). This makes algorithm development possible without needing to modify RLlib source code.
    - Refactor algorithms to match.
- [x] Refactor policy evaluators to be generic parallel samplers that are not coupled with individual algorithms. This goes with the user defined policy change.
    - Support vectorized policy evaluation for performance.
- [x] More general support for multiagent:
   - Support multiple policies in the same environment. The policies may be completely independent. Alternatively, a multi-agent policy could control multiple agents in the environment.
   - Support multiple policy optimizers operating over the same set of policy evaluators. For example, one policy may be trained with DQN, another with A3C.
- [x] General support for recurrent policies
- [ ]  Batch norm

**Comment by mathpluscode** [comment]
Hi, @ericl , thanks a lot for this amazing package. While waiting for the batch normalization feature, could you suggest a way to add this by myself ? Thank you in advance :)

**Comment by ericl** [comment]
Hey @mathpluscode, there's some discussion on adding batch norm for PPO here: https://github.com/ray-project/ray/issues/2023

Any help or ideas on how to support this more generally would be appreciated!

**Comment by ericl** [comment]
The only issue remaining is batchnorm, which should be implementable in a general way in TFPolicyGraph now.
